services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/BinaryHeadersCodec.java

```diff
@@ -0,0 +1,90 @@
+package io.scalecube.services.transport.api;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Simple binary codec for headers service message.
+ */
+public class BinaryHeadersCodec implements HeadersCodec {
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String contentType() {
+    return "application/octet-stream";
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void encode(OutputStream stream, Map<String, String> headers) throws IOException {
+    if (headers.isEmpty()) {
+      return;
+    }
+    ByteBuf buffer = ByteBufAllocator.DEFAULT.buffer();
```

Comment:
Why do you need the netty buffer in this mapper? And its allocation? And then release it?

Reply:
@segabriel https://github.com/scalecube/scalecube-services/pull/618#discussion_r299196401